### ISSUE TITLE
CobsAccumulator::feed does not allow for types with references

### ISSUE THREAD

**Issue by CaptainMaso** [jamesmunns/postcard#67]
The accumulator struct does not allow for zero copy types that reference data stored within the accumulator.
For the following code snippet:

``` Rust
#[derive(PartialEq,Eq,serde::Serialize,serde::Deserialize)]
    struct Test<'a> {
        a : &'a str
    }

    #[test]
    pub fn main() {
        let test = Test {
            a : "This is a very long string to test the data chunking of the accumulator"
        };

        let mut data = postcard::to_allocvec_cobs(&test).unwrap();
        data.extend_from_slice(&[123,123,123,123,123,123,123,123,123,123,123,123,123,123,123,0]);
        let mut data = &data[..];

        let mut acc = crate::CobsAccumulator::<512>::new();

        let deser : Test = loop {
            let mut chunk = [0;16];
            let len = data.read(&mut chunk).unwrap();

            match acc.feed(&chunk[0..len])
            {
                postcard::accumulator::FeedResult::Consumed => continue,
                postcard::accumulator::FeedResult::OverFull(e) => println!("Full"),
                postcard::accumulator::FeedResult::DeserError(e) => println!("Error"),
                postcard::accumulator::FeedResult::Success { data, remaining } => break data,
            };
        };

        assert!(deser == test);        
    }
```

The error is:
``` Rust
error: implementation of `Deserialize` is not general enough
   --> src/lib.rs:149:23
    |
149 |             match acc.feed(&chunk[0..len])
    |                       ^^^^ implementation of `Deserialize` is not general enough
    |
    = note: `Test<'_>` must implement `Deserialize<'0>`, for any lifetime `'0`...
    = note: ...but `Test<'_>` actually implements `Deserialize<'1>`, for some specific lifetime `'1`
````

I have played around and have something that might work?
Changing the lifetime from for<'de> to an explicit 'de lifetime on the function allows the returned struct to live for as long as self is mutably borrowed, meaning we can't add more data to the accumulator until we have dropped the returned struct, and the returned struct 

i.e.
``` Rust
/// Appends data to the internal buffer and attempts to deserialize the accumulated data into
/// `T`.
pub fn feed<'a,'de, T>(&'de mut self, input: &'a [u8]) -> postcard::accumulator::FeedResult<'a, T>
where
    T: serde::Deserialize<'de>,
{
    if input.is_empty() {
        return postcard::accumulator::FeedResult::Consumed;
    }

    let zero_pos = input.iter().position(|&i| i == 0);

    if let Some(n) = zero_pos {
        // Yes! We have an end of message here.
        // Add one to include the zero in the "take" portion
        // of the buffer, rather than in "release".
        let (take, release) = input.split_at(n + 1);

        // Does it fit?
        if (self.idx + n) <= N {
            // Aw yiss - add to array
            self.extend_unchecked(take);

            let retval = match postcard::from_bytes_cobs::<T>(&mut self.buf[..self.idx]) {
                Ok(t) => postcard::accumulator::FeedResult::Success {
                    data: t,
                    remaining: release,
                },
                Err(_) => postcard::accumulator::FeedResult::DeserError(release),
            };
            self.idx = 0;
            retval
        } else {
            self.idx = 0;
            postcard::accumulator::FeedResult::OverFull(release)
        }
    } else {
        // Does it fit?
        if (self.idx + input.len()) > N {
            // nope
            let new_start = N - self.idx;
            self.idx = 0;
            postcard::accumulator::FeedResult::OverFull(&input[new_start..])
        } else {
            // yup!
            self.extend_unchecked(input);
            postcard::accumulator::FeedResult::Consumed
        }
    }
}
```

**Comment by jamesmunns** [comment]
Hey! I would be open to a PR for this! I'm not *sure* if it would be a breaking change, but I'm happy to review a PR for this, and suggest alternatives (a v2 version of the type, a separate feed_v2 function with different bounds/result type, etc).

**Comment by CaptainMaso** [comment]
Fixed by #70
